homeassistant/components/history/__init__.py

```diff
@@ -83,14 +84,17 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the history hooks."""
     conf = config.get(DOMAIN, {})
 
-    filters = sqlalchemy_filter_from_include_exclude_conf(conf)
+    hass.data[HISTORY_FILTERS] = filters = sqlalchemy_filter_from_include_exclude_conf(
```

Comment:
is this expensive? Would we want to do it on demand first time the API is queried ?

Reply:
Yes the construction is expensive, we very much only want to do this once.